src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -156,6 +158,57 @@ private CentralPackageVersion ToCentralPackageVersion(string packageId, string v
             return new CentralPackageVersion(packageId, VersionRange.Parse(version));
         }
 
+        private RestoreAuditProperties GetRestoreAuditProperties()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            string enableAudit = _vsProjectAdapter.BuildProperties.GetPropertyValue(ProjectBuildProperties.NuGetAudit);
+            string auditLevel = _vsProjectAdapter.BuildProperties.GetPropertyValue(ProjectBuildProperties.NuGetAuditLevel);
+            string auditMode = _vsProjectAdapter.BuildProperties.GetPropertyValue(ProjectBuildProperties.NuGetAuditMode);
+            HashSet<string> suppressedAdvisories = GetSuppressedAdvisories();
+
+            return new RestoreAuditProperties()
+            {
+                EnableAudit = enableAudit,
+                AuditLevel = auditLevel,
+                AuditMode = auditMode,
+                SuppressedAdvisories = suppressedAdvisories,
+            };
```

Comment:
In the existing code that is deleted from this PR in favor of this method, restore audit properties are only returned when audit is enabled and audit level is set. If these properties are not set then `null` is returned. However, in this case my understanding is that we always return `RestoreAuditProperties`. I am assuming that should be okay because if audit is disabled these properties will have their values configured accordingly but just want to check.

```cs
RestoreAuditProperties auditProperties = !string.IsNullOrEmpty(enableAudit) || !string.IsNullOrEmpty(auditLevel)
                ? new RestoreAuditProperties()
                {
                    EnableAudit = enableAudit,
                    AuditLevel = auditLevel,
                    AuditMode = auditMode,
                }
                : null;
```

Reply:
A few things:

* This ensures no-op consistency across VS/CLI. 
* The NuGet.Targets file has default, so in practice those values will never be empty: https://github.com/NuGet/NuGet.Client/blob/252bbee32a839623dc9af267f950d21cde5e9497/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L70-L77